enforcer-rules/src/main/java/org/apache/maven/plugins/enforcer/BannedPlugins.java

```diff
@@ -29,12 +28,13 @@
  */
 public class BannedPlugins extends BannedDependencies {
     @Override
-    protected Set<Artifact> getDependenciesToCheck(ProjectBuildingRequest buildingRequest) {
-        return buildingRequest.getProject().getPluginArtifacts();
```

Comment:
This is important ... each extended rule must provided a dependency to check in BannedDependencies

Reply:
No, it doesn't work based on artifact sets anymore.

Reply:
Apologies for that, let me check why it's failing and I'll create more tests once done.

Reply:
Corrected that, and added a simple test on banned dependencies failure.

Reply:
please approve the flow 🤡